src/azure-cli/azure/cli/command_modules/monitor/operations/log_alert.py

```diff
@@ -0,0 +1,252 @@
+                                                                                           resource_group_name))
+
+    if metric_threshold or metric_column or metric_trigger_type or metric_threshold_operator:
+        metric_trigger = LogMetricTrigger(metric_trigger_type=metric_trigger_type, metric_column=metric_column,
+                                          threshold_operator=metric_threshold_operator, threshold=metric_threshold)
+        trigger = TriggerCondition(threshold_operator=threshold_operator,
+                                   threshold=threshold, metric_trigger=metric_trigger)
+    else:
+        trigger = TriggerCondition(threshold_operator=threshold_operator, threshold=threshold)
+
+    if action_group:
+        action_group = _normalize_names(cmd.cli_ctx, action_group, resource_group_name, 'microsoft.insights',
+                                        'actionGroups')
+
+    action = AlertingAction(severity=severity, trigger=trigger, throttling_in_min=throttling,
+                            azns_action=AzNsActionGroup(action_group=action_group, email_subject=email_subject,
+                                                        custom_webhook_payload=None))
+
+    settings = LogSearchRuleResource(location=location, tags=tags, description=description, enabled=not disable,
+                                     source=Source(query_type=query_type, data_source_id=data_source_id,
+                                                   query=alert_query, authorized_resources=authorized_resources),
+                                     schedule=Schedule(frequency_in_minutes=frequency,
+                                                       time_window_in_minutes=time_window), action=action)
+    return client.create_or_update(resource_group_name, rule_name, settings)
+
+
+def list_log_alert(client, resource_group_name=None):
+    if resource_group_name:
+        return client.list_by_resource_group(resource_group_name)
+
+    return client.list_by_subscription()
+
+
+def update_log_alert(  # pylint: disable=too-many-locals
+        cmd, instance, resource_group_name, enabled=None, tags=None, description=None, frequency=None,
+        time_window=None, alert_query=None, severity=None,
+        threshold_operator=None, threshold=None, throttling=None,
+        metric_column=None, metric_trigger_type=None, metric_threshold_operator=None, metric_threshold=None,
+        reset_action_group=None, add_action_groups=None, remove_action_groups=None,
+        email_subject=None, reset_email_subject=None,
+        reset_metric_trigger=None, reset_authorized_resources=None,
+        add_authorized_resources=None, remove_authorized_resources=None):
+    # --tags "" is set as tags={}. Used for clearing tags.
+    if tags or tags == {}:
+        instance.tags = tags
+
+    if description:
+        instance.description = description
+
+    if frequency:
```

Comment:
Please be careful about such `instance.xxx.xxx`. Some properties are optional and the server will not return a complete response. `instance.xxx.xxx` will be None type and this will raise error.

Reply:
You might confirm the back-end server behavior about these fileds

Reply:
Handled cases for optional parameters.